### ISSUE TITLE
Import from CSV

### ISSUE THREAD

**Issue by gagarine** [mstarke/MacPass#629]
It's possible to convert lastpass csv to xml: https://github.com/srvz/lastpass-to-macpass

But this require developer knowledge (terminal, nodejs, ..).

Would be nice to have a CSV import for lastpass, dashlane and 1password exported files directly  integrated in MacPass.

**Comment by mstarke** [comment]
CVS Import is a common feature request and should be tackled once the core KeePass features are stable an present.

**Comment by tflo** [comment]
Just for info: The current version of KeePassXC does CSV import pretty well. I successfully imported from LastPass and 1Password.

Since this is a feature you need maybe once every couple of years, or even only once in your life, I would be pleased to not see the development of MacPass *focusing* on that feature. In the meantime it should be legit to “abuse” KeePassXC solely for that feature ;)

– Tom

**Comment by gagarine** [comment]
@tflo I do not agree because user experience is reduced by a lot for new users. Most users have a hard time to install a program or even make the difference between a file and a program... Yet it's exactly those users that manage their password very badly. The value of a password manager is in the UX. Otherwise one can simply use a txt file in an encrypted image... ;)

This is one of the problems of open source software -> You don't have that much incentive to invest that much in reducing friction for new users.

**Comment by neo-anderson** [comment]
@tflo Thanks for the KeepassXC tip! I wish there was an import csv option in MacPass too. Took me a while to figure out how to get all the data from LastPass into MacPass

**Comment by mstarke** [comment]
I guess a CSV importer should be upped on the list :)

**Comment by sebknzl** [comment]
Just discovered MacPass today, thanks a lot for the nice work!

About CSV import, I've used this as a workaround:

 * Use https://github.com/lifepillar/csv2keepassxml to convert the CSV to XML
 * Create a new database and import the XML

Once the database is saved, you can also merge it to an existing database.

**Comment by stam** [comment]
Would like this as well! Am moving from Enpass to Macpass. Enpass supports csv, json and txt export, but Macpass only supports an xml import

**Comment by mstarke** [comment]
I've started a CSV import/export plugin project https://github.com/MacPass/MacPassCSV

Currently only export works. Import is about to be added. If anyone wants to contribute, I'm more than happy to merge an PRs.

I need to check what CSV flavours are commonly exported by all password clients but it might get a bit tricky. Is there are comprehensive list of CSV exporting password clients and their format?

**Comment by mstarke** [comment]
Thanks for the tip. When I was researching for this plugin some time ago I came across various CSV parser libraries but from what I have gathered so fart the CSV I/O part is the least problematic. Parsing the format in the correct way and constructing a reasonable tree from the import is the real challenge IMHO.

**Comment by maltokyo** [comment]
Just import into KeePassXC, and then open the resulting DB with MacPass, no?  Easy.

**Comment by mstarke** [comment]
I've started a plugin (https://github.com/MacPass/MacPassCSV) to get this off the ground but did not finish it. Need a lot more work but it might be picked up by the community (or me :) ).
